app/src/androidTest/java/com/github/se/orator/endtoend/EndToEndTest.kt

```diff
@@ -250,5 +356,159 @@ class EndToEndAppTest {
 
     // Step 7: Navigate back to Home from Friends
     composeTestRule.onNodeWithTag("Home").performClick() // Simulate clicking Home button
+
+    `when`(speakingRepository.analysisState)
+        .thenReturn(MutableStateFlow(SpeakingRepository.AnalysisState.IDLE))
+
+    // Step 8: Speaking screen
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.SPEAKING) // forcing back to home where the user would be once he finishes creating
+      // his
+      // account
+    }
+    composeTestRule.onNodeWithTag("ui_column").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("back_button").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("mic_text").assertTextContains("Tap the mic to start recording.")
+
+    composeTestRule.onNodeWithTag("mic_button").assertIsDisplayed()
+    composeTestRule.onNodeWithContentDescription("Start recording").assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("ui_column").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("back_button").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("back_button").performClick()
+
+    verify(navigationActions).goBack()
+    clearInvocations(navigationActions)
+
+    `when`(speakingRepository.analysisState)
+        .thenReturn(MutableStateFlow(SpeakingRepository.AnalysisState.RECORDING))
+
+    speakingViewModel =
+        SpeakingViewModel(speakingRepository, apiLinkViewModel, userProfileViewModel)
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.SPEAKING) // forcing back to home where the user would be once he finishes creating
+      // his account
+    }
+    composeTestRule.onNodeWithTag("mic_text").assertTextContains("Recording...")
+
+    composeTestRule.onNodeWithTag("mic_button").assertIsDisplayed()
+    composeTestRule.onNodeWithContentDescription("Stop recording").assertIsDisplayed()
+
+    composeTestRule.onNodeWithTag("mic_button").performClick()
+    //    composeTestRule.onNodeWithTag("mic_text").assertTextContains("Analysis finished.")
+    composeTestRule.onNodeWithTag("mic_button").assertIsDisplayed()
+
+    composeTestRule.onNodeWithText("Transcribed Text: $speech").assertExists()
+    composeTestRule.onNodeWithText("Sentiment Analysis: 1.0").assertIsDisplayed()
+
+    `when`(speakingRepository.analysisState)
+        .thenReturn(MutableStateFlow(SpeakingRepository.AnalysisState.FINISHED))
+
+    speakingViewModel =
+        SpeakingViewModel(speakingRepository, apiLinkViewModel, userProfileViewModel)
+
+    composeTestRule.runOnUiThread {
+      navController?.navigate(
+          Screen.SPEAKING) // forcing back to home where the user would be once he finishes creating
+      // his account
```

Comment:
Not sure this comment should be here.
```suggestion
          Screen.SPEAKING)
```

Reply:
oops I just copied the code to forcibly move to another screen! Great catch.